src/hooks/useLeaderboard.tsx

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react';
+import {
+  LeaderboardType,
+  LeaderboardDataHandler,
+  AllLeaderboardData,
+  AllOrderings,
+} from '../types';
+import { useToast } from '@chakra-ui/react';
+import {
+  getGitHubLeaderboard,
+  getInternshipLeaderboard,
+  getLeetcodeLeaderboard,
+  getNewGradLeaderboard,
+} from '../services/leaderboard';
+import { assertTypeAndOrderingIntegrity } from '../utils';
+
+const getLeaderboardBasedOnType = (
```

Comment:
nit: name is a little too vague to guess that this is doing

Reply:
Fixed